templates/settings.php

```diff
@@ -34,7 +34,8 @@
 	$this->options['track-logged-in']     = ( $_POST['track-logged-in'] == "no" ) ? false : true;
 	$this->options["untrack-roles"]       = ! empty( $_POST["tracker_role"] ) ? array_map( 'esc_attr', $_POST["tracker_role"] ) : "";
 	$this->options["widget-access-roles"] = ! empty( $_POST["widget_access"] ) ? array_map( 'esc_attr', $_POST["widget_access"] ) : "";
-	$this->options["tracker-address"] = ! empty( $_POST["tracker-address"] ) ? esc_url_raw( $_POST['tracker-address'] ) : "https://mc.yandex.ru/metrika/watch.js";
+	$this->options['new_yandex_code']     = empty( $_POST['new_yandex_code'] ) ? false : true;
+	$this->options["tracker-address"]     = ! empty( $_POST["tracker-address"] ) ? esc_url_raw( $_POST['tracker-address'] ) : ($this->options['new_yandex_code'] ? "https://mc.yandex.ru/metrika/tag.js" : "https://mc.yandex.ru/metrika/watch.js");
```

Comment:
avoid short if/else controls if there is a complicated situation (better readability)